src/Tickets/Commerce/Gateways/PayPal/WhoDat.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\PayPal;
+
+/**
+ * Class Connect_Client
+ *
+ * @since   5.1.6
+ *
+ * @package TEC\Tickets\Commerce\Gateways\PayPal
+ */
+class WhoDat {
+
+	/**
+	 * The API URL.
+	 *
+	 * @since 5.1.6
+	 *
+	 * @var string
+	 */
+	protected $api_url = 'https://whodat.theeventscalendar.com/tickets/paypal/connect';
+
+	/**
+	 * Get REST API endpoint URL for requests.
+	 *
+	 * @since TBD
+	 *
+	 *
+	 * @param string $endpoint   The endpoint path.
+	 * @param array  $query_args Query args appended to the URL.
+	 *
+	 * @return string The API URL.
+	 */
+	public function get_api_url( $endpoint, array $query_args = [] ) {
+		return add_query_arg( $query_args, "{$this->api_url}/{$endpoint}" );
+	}
+
+	/**
+	 * Send a POST request to WhoDat inside of the PayPal connection path.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $endpoint
+	 * @param array  $query_args
+	 * @param array  $request_arguments
+	 *
+	 * @return array|null
+	 */
+	public function post( $endpoint, array $query_args = [], array $request_arguments = [] ) {
+		$url = $this->get_api_url( $endpoint, $query_args );
+
+		$default_arguments = [
+			'body'      => [],
+
+			// @todo Remove this when SSL is fixed.
+			'sslverify' => false,
+		];
+		$request_arguments = array_merge_recursive( $default_arguments, $request_arguments );
+		$response          = wp_remote_post( $url, $request_arguments );
+
+		if ( is_wp_error( $response ) ) {
+			tribe( 'logger' )->log_error( sprintf(
+				'[%s] WhoDat request error: %s',
+				$url,
+				$response->get_error_message()
+			), 'whodat-connection' );
+
+			return null;
+		}
+
+		$response = wp_remote_retrieve_body( $response );
+		$response = @json_decode( $response, true );
+
+		if ( ! is_array( $response ) ) {
+			tribe( 'logger' )->log_error( sprintf( '[%s] Unexpected WhoDat response', $url ), 'whodat-connection' );
+
+			return null;
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves a Partner Link for on-boarding
+	 *
+	 * @param $return_url
+	 * @param $country
+	 *
+	 * @return array|null
+	 */
+	public function get_seller_partner_link( $return_url, $country ) {
+		$query_args = [
+			'mode' => tribe( Merchant::class )->get_mode(),
+			'request' => 'partner-link',
+		];
+
+		$args = [
+			'body' => [
+				'return_url'   => $return_url,
+				'country_code' => $country,
+			]
+		];
+
+		return $this->post( 'paypal-commerce', $query_args, $args );
+	}
+
+	/**
+	 * Get seller on-boarding details from seller.
+	 *
+	 * @since 5.1.6
+	 *
+	 * @param string $access_token
+	 * @param string $merchant_id
+	 *
+	 * @return array|null
+	 */
+	public function get_seller_on_boarding_details_from_paypal( $merchant_id, $access_token ) {
```

Comment:
Same as previous comment: would recommend ditching based on our convo earlier today.

Reply:
Using the code linked below as an example:
https://github.com/the-events-calendar/event-tickets/pull/2039/files#diff-bfb8373a3b32358183c486ebe955164d7f5df080d3eb51c0f3fafba370620a72R89-R99

We should create a method on the `WhoDat` class to introduce all the HTTP Request related business, and on the `Onboard` class we should do only the parsing of that data to determine for example the status.

Meaning only the lines below would be part of onboard, and the ones before that would be part of WhoDat

https://github.com/the-events-calendar/event-tickets/pull/2039/files#diff-bfb8373a3b32358183c486ebe955164d7f5df080d3eb51c0f3fafba370620a72R101-R109

Reply:
Ref. the HTTP Request > agreed: this is very well aligned with what was previously mentioned [here](https://github.com/the-events-calendar/event-tickets/pull/2036#discussion_r676811351). 

#2039 is a draft PR, meaning work is still in progress: it would be great to have recommendations and changes to the code happen in the context of that work (as soon as its complete), rather than a refactor in the base branch.